### PR TITLE
sigtop: init at 0.3.1

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
sigtop is a utility to export messages, attachments and other data from Signal Desktop.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/backup/sigtop/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  makeFlags = [
+    "PREFIX=\${out}"
+  ];
+
+  meta = with lib; {
+    description = "Utility to export messages, attachments and other data from Signal Desktop";
+    license = licenses.isc;
+    maintainers = with maintainers; [ fricklerhandwerk ];
+  }
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/sigtop/default.nix:25]:
```suggestion
  };
```

File: pkgs/tools/backup/sigtop/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitHub {
+    owner = "tbvdm";
+    repo = "sigtop";
+    # `portable` branch
+    rev = "069dd4d0526a795c00eb34dd1112637f6d8994a4";
+    sha256 = "1gmxyjx7kbr2n33aq84ipa2pfhrdw7v7ggz01qd54nqyvp8hdbzs";
+  };
+
+  buildInputs = [ openssl pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/sigtop/default.nix:15]: pkg-config goes to nativeBuildInputs.

File: pkgs/tools/backup/sigtop/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchFromGitHub, openssl, pkg-config }:
+
+stdenv.mkDerivation {
+  name = "sigtop";
+  version = "2022-03-21";
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/sigtop/default.nix:5]:
```suggestion
  version = "unstable-2022-05-27";
```

File: pkgs/tools/backup/sigtop/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitHub {
+    owner = "tbvdm";
+    repo = "sigtop";
+    # `portable` branch
+    rev = "069dd4d0526a795c00eb34dd1112637f6d8994a4";
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/sigtop/default.nix:11]:
```suggestion
    rev = "945c5844d25e2b130809334cbc8f3fa1cd85aaf9";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Missing top-level entry

**Comment by fricklerhandwerk** [comment]
@SuperSandro2000 Thanks a lot for checking!

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please follow the contributing guide when creating your commits.

File: pkgs/tools/backup/sigtop/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+stdenv.mkDerivation {
+  name = "sigtop";
+  version = "unstable-2022-05-27";
+
+  src = fetchFromGitHub {
+    owner = "tbvdm";
+    repo = "sigtop";
+    # `portable` branch
+    rev = "945c5844d25e2b130809334cbc8f3fa1cd85aaf9";
+    sha256 = "1gmxyjx7kbr2n33aq84ipa2pfhrdw7v7ggz01qd54nqyvp8hdbzs";
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/sigtop/default.nix:12]:
```suggestion
    sha256 = "sha256-6Ifs8fsNpE3AstNt3Elyy908sQ4JzE1H8DbUeWhGnLE=";
```

File: pkgs/tools/backup/sigtop/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  buildInputs = [ openssl ];
+
+  nativeBuildInputs = [ pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/sigtop/default.nix:17]:
```suggestion
  nativeBuildInputs = [ pkg-config ];

  buildInputs = [ openssl ];
```

File: pkgs/tools/backup/sigtop/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  buildInputs = [ openssl ];
+
+  makeFlags = [
+    "PREFIX=\${out}"
+  ];
+
+  meta = with lib; {
+    description = "Utility to export messages, attachments and other data from Signal Desktop";
+    license = licenses.isc;
+    maintainers = with maintainers; [ fricklerhandwerk ];
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/sigtop/default.nix:26]: You should set platforms.

**Pull request merged**